### PR TITLE
feat: using react-hook-form for form handling and tanstack query for …

### PR DESCRIPTION
**Pull request by shanksxz**
### Summary:
This pull request introduces **React Hook Form** for handling forms and **TanStack Query** for managing data fetching and mutations.

### Key Changes:
1. **React Hook Form** has been implemented to manage form state and validation, reducing re-renders and improving performance.
2. **TanStack Query** has been integrated for efficient server state management, handling data fetching and mutations with optimized performance and caching.

### Impact:
- **Form Handling**: React Hook Form simplifies validation and state management, providing a more responsive form experience.
- **Data Fetching & Mutations**: TanStack Query improves the efficiency of asynchronous operations, including data fetching, caching, and updates.

### Related Issue:
Resolves issue #28

### CODE REVIEW

**Comment by yesahem** [comment]
@geekyharsh05 Please review it!!!


I have no idea about tanstack querry (learning)

File: frontend/src/routes/Blogs.tsx [hunk 1]
```diff
@@ -16,91 +17,87 @@ interface posts {
...
-        console.log(err);
-      });
-  }, [description2, token]);
-
   const truncateContent = (content: string, length: number) => {
     if (content.length <= length) return content;
     return content.substring(0, length) + "...";
   };
 
+  if(foo || isError) {
```
Comment (geekyharsh05) [inline frontend/src/routes/Blogs.tsx]: Can you please change the name of this to some meaningful names.
like 

```bash
 if (postError || userError) 
```

File: frontend/src/routes/UploadBlogs.tsx [hunk 2]
```diff
@@ -1,52 +1,51 @@
...
       console.log("Response after blog upload:", res);
-      // Redirect to the blog page after successful upload
-      navigate("/blog"); // Adjust the path as needed
-    } catch (err) {
-      console.error("Error uploading blog:", err);
-    }
+      navigate("/blog");
+    },
+    onError: (error) => {
+      console.error("Error uploading blog:", error);
```
Comment (geekyharsh05) [inline frontend/src/routes/UploadBlogs.tsx:38]: Please add a toast message here.
```bash
alertError("Error uploading the blog. Please try again.");
```

File: frontend/src/routes/UploadBlogs.tsx [hunk 3]
```diff
@@ -1,52 +1,51 @@
...
           published: true,
         },
         {
           headers: {
             Authorization: `Bearer ${token}`,
           },
         }
-      );
+      ),
+    onSuccess: (res) => {
```
Comment (geekyharsh05) [inline frontend/src/routes/UploadBlogs.tsx:32]: Please add a success toast message here.
```bash
alertSuccess();
```

**Comment by geekyharsh05** [comment]
@shanksxz Great Work!

File: frontend/src/routes/Blogs.tsx (hunk 1 above)
Comment (shanksxz) [reply frontend/src/routes/Blogs.tsx]: done

File: frontend/src/routes/UploadBlogs.tsx (hunk 2 above)
Comment (shanksxz) [reply frontend/src/routes/UploadBlogs.tsx:38]: done

**Comment by geekyharsh05** [comment]
@shanksxz LGTM! Thank you for your contribution!

**Comment by yesahem** [comment]
@geekyharsh05 bhaiiiiii !!!!

**Pull request merged**